src/bbs2gh/Handlers/MigrateRepoCommandHandler.cs

```diff
@@ -88,6 +89,15 @@ public async Task Handle(MigrateRepoCommandArgs args)
         }
     }
 
+    private string GetSourceExportArchiveAbsolutePath(string bbsSharedHomeDirectory, long exportId)
+    {
+        static string GetDefaultBbsSharedHomeDirectory() => RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
+            ? BbsSmbArchiveDownloader.DEFAULT_BBS_SHARED_HOME_DIRECTORY
+            : BbsSshArchiveDownloader.DEFAULT_BBS_SHARED_HOME_DIRECTORY;
+
```

Comment:
Wouldn't this function be better living in IBbsArchiveDownloader?

Reply:
Well actually a good idea, I can create two constant one for windows and one for linux

Reply:
Might be a good argument for having this as a base class instead of an interface. Then you could have a virtual `DEFAULT_BBS_SHARED_HOME_DIRECTORY` that you overload in each subclass.

Having an interface (or base class) have to "know" about every subclass/implementation seems bad.

Reply:
I still like an interface better. The problem is not overriding the default shared directory since it can be easily done with a property in the interface too. The problem is that I need to access the constants outside of the concrete implementations too, in our case in the handler class. Basically depending on the OS BBS has two default home directories.
I originally took the constants out of the interface and put them into the concrete implementations because as you also mentioned I didn't want the interface to know about the implementations. But the ultimate problem is that I want to have these constants defined only in one place. So I have the following options now:
1. Leave everything and access the default shared home through a concrete downloader implementation. 
2. Create a class to hold these constants (e.g. `BbsSettings`, `Defaults`, ...).
3. Move the constants back into the interface and rename them to indicate one is for linux and the other for windows.
4. Create a service (e.g. `BbsDefaultHomeDirectoryProvider`) --> this is an overkill

Now with everything being said, I am personally leaning toward option 1 and 3. What do you think ?

Reply:
Also as a quick side note, using ssh doesn't necessarily mean that the BBS's OS is linux! Ppl can opt-in to use ssh on a windows server too.